### PR TITLE
Fix duplicated records in avatax for confirmed orders

### PR DESCRIPTION
**Pull request by IKarbowiak**
When an order is confirmed we want to attach only an undiscounted amount to request data for avatax.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-5638 Duplicate records of lines in Avatax](https://app.clickup.com/t/2549495/SALEOR-5638)

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

File: saleor/plugins/avatax/__init__.py [hunk 1]
```diff
@@ -360,24 +360,29 @@ def get_order_lines_data(
             amount=undiscounted_amount,
         )
 
-        if undiscounted_amount != price_with_discounts_amount:
+        # for invoice transaction we want to include only undiscounted price
```
Comment (korycins) [inline saleor/plugins/avatax/__init__.py:363]: Are you sure of this description? I thought that for invoice transactions we want to only include a "final" price, like price_with_discounts

File: saleor/plugins/avatax/__init__.py [hunk 2]
```diff
@@ -360,24 +360,29 @@ def get_order_lines_data(
...
-        append_line_to_data(
-            data=data,
-            quantity=1,
-            amount=discount_amount.amount * -1,
-            tax_code=COMMON_DISCOUNT_VOUCHER_CODE,
-            item_code="Voucher",
-            name="Order discount",
-            tax_included=True,  # Voucher should be always applied as a gross amount
-        )
+    if not invoice_transaction_type:
```
Comment (korycins) [inline saleor/plugins/avatax/__init__.py:374]: Are you sure of this? There is still a possibility that we add a voucher to the entire order. This voucher will be included here. In previous approach we send this to avatax (which was OK)

Comment (IKarbowiak) [reply saleor/plugins/avatax/__init__.py]: Good point.

**Pull request merged**